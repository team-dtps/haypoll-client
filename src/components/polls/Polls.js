import React from 'react';
import PropTypes from 'prop-types';

export default function Polls({ polls }) {
  const pollItems = polls.map(poll => {
    return <li key={poll._id}>poll.questions</li>;
  });

  return (
    <ul> {pollItems} </ul>
  );
}

Polls.propTypes = {
  polls: PropTypes.array.isRequired
};
