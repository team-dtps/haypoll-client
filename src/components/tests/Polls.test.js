import React from 'react';
import renderer from 'react-test-renderer';
import Polls from '../polls/Polls';

describe('Polls snapshot', () => {
  it('can match a snapshot', () => {
    const polls = [ 
      { question: 'Poll question', choices: ['choice1', 'choice2', 'choice3'], creator: 'voter1' }
    ];
    const tree = renderer.create(
      <Polls polls={polls}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

