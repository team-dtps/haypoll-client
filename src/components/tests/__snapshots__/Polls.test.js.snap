// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polls snapshot can match a snapshot 1`] = `
<ul>
   
  <li>
    poll.questions
  </li>
   
</ul>
`;
